src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/DownloadManager.java

```diff
@@ -123,9 +124,13 @@ public Future<Path> startDownload(
       Path output,
       ExtendedEventHandler eventHandler,
       Map<String, String> clientEnv,
-      String context) {
+      String context,
+      Phaser downloadPhaser) {
     return executorService.submit(
         () -> {
+          if (downloadPhaser.register() < 0) {
```

Comment:
```suggestion
          if (downloadPhaser.register() != 0) {
```
looks more obviously correct to me, although I am not 100% sure it even makes a difference.